extensions/wikia/CuratedContent/CuratedContentController.class.php

```diff
@@ -599,32 +614,78 @@ private function sumUpQualityStats( $stats ) {
 	}
 
 	/**
-	 * @param array $curatedContent curated sections data
-	 * @return array
+	 * @param array $section curated content single section data
+	 * @return array section extended with node type, image data and items type
 	 */
-	private function extendSections( $curatedContent ) {
-		$validator = new CuratedContentValidator();
+	private function extendSectionAndItems( $section ) {
+		$section['node_type'] = 'section';
 
-		$data = array_map( function ( $section ) use ( $validator ) {
-			$featured = isset( $section['featured'] );
-			$section['node_type'] = 'section';
-			$section['items'] = array_values(
-				array_filter( $section['items'], function ( $item ) use ( $validator, $featured ) {
-					$error = $featured ? $validator->validateFeaturedItem( $item ) :
-						$validator->validateSectionItem( $item );
-					return empty( $error );
-				} )
-			);
+		if ( !empty( $section['items'] ) ) {
 			$section['items'] = $this->extendItemsWithImages( $section['items'] );
 			$section['items'] = $this->extendItemsWithType( $section['items'] );
-			return $section;
-		}, $curatedContent );
+		}
+
+		return $section;
+	}
+
+	/**
+	 * @param array $section featured categories section data
+	 * @return array input data (full section return only if valid) with validated items only (featured categories).
+	 */
+	private function validateCuratedSection( $section ) {
+		if ( empty( $section['items'] ) ) {
+			return [];
+		}
+
+		$validator = new CuratedContentValidator();
+		$error = $validator->validateSection( $section );
 
-		$data = array_values( array_filter( $data, function ( $section ) use ( $validator ) {
-			$error = isset( $section['featured'] ) ? false : $validator->validateSection( $section );
+		if ( !empty( $error ) ) {
+			return [];
+		}
+
+		$section['items'] = array_filter( $section['items'], function ( $item ) use ( $validator ) {
+			$error = $validator->validateSectionItem( $item );
+			return empty( $error );
+		} );
+
+		return $section;
+	}
+
+	/**
+	 * @param array $section featured content section data
+	 * @return array input data but with validated items only (featured content slide).
+	 */
+	private function validateFeaturedSection( $section ) {
+		if ( empty( $section['items'] ) ) {
+			return [];
+		}
+
+		$validator = new CuratedContentValidator();
+		$section['items'] = array_filter( $section['items'], function ( $item ) use ( $validator ) {
+			$error = $validator->validateFeaturedItem( $item );
 			return empty( $error );
-		} ) );
-		return $data;
+		} );
+
+		return $section;
+	}
+
+	/**
+	 * @param array $section optional categories section data
+	 * @return array input data with validated items only (optional categories).
+	 */
+	private function validateOptionalSection( $section ) {
```

Comment:
private function validateOptionalSection( array $section ): array {}
please add types to functions